state/src/caching/order_metadata_index.rs

```diff
@@ -114,6 +118,10 @@ impl OrderMetadataIndex {
             sorted_vec.remove(idx);
         }
 
+        // Remove from the reverse index
+        let mut reverse_index = self.reverse_index.write().await;
+        reverse_index.remove(order_id);
+
```

Comment:
did we want to move this back under the fallible code?

Reply:
yes, the code above it depends on the entry still being in place